messaging/kafkaSSL/src/main/java/io/quarkus/qe/messaging/ssl/quickstart/KafkaEndpoint.java

```diff
@@ -0,0 +1,52 @@
+package io.quarkus.qe.messaging.ssl.quickstart;
+
+import java.time.Duration;
+import java.util.Collections;
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import org.apache.kafka.clients.admin.AdminClient;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.clients.producer.ProducerRecord;
+
+public abstract class KafkaEndpoint {
+
+    public static final String TOPIC = "hello";
+    volatile boolean done = false;
+    volatile String last;
+
+    protected void initialize(KafkaConsumer<String, String> consumer) {
+        consumer.subscribe(Collections.singleton(TOPIC));
+        new Thread(() -> {
+            while (!done) {
```

Comment:
I can't find the place where done is set to TRUE, can you give me a hint, please?
FULL DISCLOSURE: my IDE is still indexing, maybe I just missed it, please kindly point me to the right place.

Reply:
This is because is an apache Kafka client. In this scenario we are using this extension:

`io.quarkus:quarkus-kafka-client` that bascially is a pure Kafka client integration. On this consumer, we are subscribing to a topic with a new thread, by polling. So, we are requesting new records during 1 sec

```
consumer.poll(Duration.ofSeconds(1));
```

This way to implement a client is quite common, take a look for example at this confluent consumer: 
https://docs.confluent.io/kafka-clients/java/current/overview.html#basic-poll-loop